src/foundry/foundry.js/applications/basePlaceableHUDs/tokenHUD.d.ts

```diff
@@ -1,137 +1,146 @@
+      canConfigure: boolean;
+      canToggleCombat: boolean;
+      displayBar1: boolean;
+      bar1Data: ReturnType<ConcreteObject['document']['getBarAttribute']>;
+      displayBar2: boolean;
+      bar2Data: ReturnType<ConcreteObject['document']['getBarAttribute']>;
+      visibilityClass: string;
+      effectsClass: string;
+      combatClass: string;
+      targetClass: string;
+      statusEffects: ReturnType<TokenHUD['_getStatusEffectChoices']>;
+    };
+
+    /**
+     * Get an array of icon paths which represent valid status effect choices
+     * @internal
+     */
+    protected _getStatusEffectChoices(): Partial<
+      Record<
+        string,
+        {
+          id: string;
+          title: string;
+          src: string;
+          isActive: boolean;
+          isOverlay: boolean;
+          cssClass: string;
+        }
+      >
+    >;
+
+    /** @override */
+    activateListeners(html: JQuery): void;
+
+    /** @override */
+    protected _onClickControl(event: JQuery.ClickEvent): Promise<void | boolean> | void;
+
+    /**
+     * Handle initial click to focus an attribute update field
+     * @internal
+     */
+    protected _onAttributeClick(event: JQuery.ClickEvent): void;
+
+    /**
+     * Force field handling on an Enter keypress even if the value of the field did not change.
+     * This is important to suppose use cases with negative number values.
+     * @param event - The originating keydown event
+     * @internal
+     */
+    protected _onAttributeKeydown(event: JQuery.KeyDownEvent): void;
+
+    /**
+     * Handle attribute bar update
+     * @internal
+     */
+    protected _onAttributeUpdate(event: JQuery.ChangeEvent): void;
+
+    /**
+     * Toggle Token combat state
+     * @internal
+     */
+    protected _onToggleCombat(event: JQuery.ClickEvent): Promise<void>;
+
+    /**
+     * Handle Token configuration button click
+     * @internal
+     */
+    protected _onTokenConfig(event: JQuery.ClickEvent): void;
+
+    /**
+     * Handle left-click events to toggle the displayed state of the status effect selection palette
+     * @internal
+     */
+    protected _onClickStatusEffects(event: JQuery.ClickEvent): void;
+
+    /**
+     * Assign css selectors for the active state of the status effects selection palette
+     * @internal
+     */
+    protected _toggleStatusEffects(active: boolean): void;
+
+    /**
+     * Handle toggling a token status effect icon
+     * @param overlay - (default: `false`)
+     * @internal
+     */
+    protected _onToggleEffect(
+      event: JQuery.ClickEvent | JQuery.ContextMenuEvent,
+      { overlay }?: { overlay?: boolean }
+    ): Promise<boolean>;
+
+    /**
+     * Handle toggling the target state for this Token
+     * @internal
+     */
+    protected _onToggleTarget(event: JQuery.ClickEvent): void;
+  }
 }
+
+type ConcreteObject = InstanceType<ConfiguredObjectClassForName<'Token'>>;
```

Comment:
Might make sense to call it `ConcreteToken` os similar, similarly for the other HUDs